### PR TITLE
Add support to Microsoft Rich Edit 4.1 and up.

### PR DESCRIPTION
**Pull request by XindongSu**
Simply added supporting to Microsoft Rich Edit 4.1 and up, which is supported through MSFTEDIT.DLL. With adding this, JvRichEdit can provide more compatibility with other document processor softwares, especialy the Microsoft Office.

No new features of Rich Edit 4.1 or up are implemented yet. 

For reference of the versions of Rich Edit Controls:
[https://docs.microsoft.com/en-us/windows/win32/controls/about-rich-edit-controls](https://docs.microsoft.com/en-us/windows/win32/controls/about-rich-edit-controls)

### CODE REVIEW

File: jvcl/run/JvRichEdit.pas [hunk 1]
```diff
@@ -51,6 +51,14 @@ interface
   Forms, Graphics, StdCtrls, Dialogs, RichEdit, Menus, ComCtrls, SyncObjs,
   JvExStdCtrls, JvTypes;
 
+const
+  {$EXTERNALSYM MSFTEDIT_CLASSA}
+  MSFTEDIT_CLASSA       = 'RICHEDIT50A';     { Richedit4.0 Window Class. }
+  {$EXTERNALSYM MSFTEDIT_CLASSW}
+  MSFTEDIT_CLASSW       = 'RICHEDIT50W';     { Richedit4.0 Window Class. }
+  {$EXTERNALSYM MSFTEDIT_CLASS}
+  MSFTEDIT_CLASS = MSFTEDIT_CLASSW;
```
Comment (obones) [inline jvcl/run/JvRichEdit.pas]: Instinctively, I believe this should be IFDED depending on whether the RTL is unicode or not. However, looking at line 2840, I even wonder if it shouldn't simply be the ANSI version

File: jvcl/run/JvRichEdit.pas [hunk 2]
```diff
@@ -7716,18 +7731,31 @@ procedure InitRichEditDll;
...
+  GLibHandle := SafeLoadLibrary(RichEdit40ModuleName);
+  if (GLibHandle > 0) and (GLibHandle < HINSTANCE_ERROR) then
+    GLibHandle := 0
+  else
+    RichEditVersion := 4; // at least version 4
+
+  if GLibHandle = 0 then
+  begin
   GLibHandle := SafeLoadLibrary(RichEdit20ModuleName);
   if (GLibHandle > 0) and (GLibHandle < HINSTANCE_ERROR) then
```
Comment (obones) [inline jvcl/run/JvRichEdit.pas]: Why was this not indented?

File: jvcl/run/JvRichEdit.pas (hunk 1 above)
Comment (ahausladen) [reply jvcl/run/JvRichEdit.pas:60]: [RichEdit 1.0](https://docs.microsoft.com/en-us/windows/win32/controls/about-rich-edit-controls#rich-edit-version-10) only supported ANSI. Starting with [RichEdit 2.0](https://docs.microsoft.com/en-us/windows/win32/controls/about-rich-edit-controls#rich-edit-version-20) Unicode was added.

File: jvcl/run/JvRichEdit.pas (hunk 2 above)
Comment (ahausladen) [reply jvcl/run/JvRichEdit.pas:7744]: I merge it and fixed the indention

**Pull request merged**